src/main/java/gtPlusPlus/core/tileentities/general/TileEntityFishTrap.java

```diff
@@ -50,23 +50,18 @@ private boolean isSurroundedByWater() {
         int waterCount = 0;
         int trapCount = 0;
         for (final Block checkBlock : surroundingBlocks) {
-            if ((checkBlock == Blocks.water) || (checkBlock == Blocks.flowing_water)
-                    || checkBlock.getUnlocalizedName().toLowerCase().contains("water")
-                    || (checkBlock == ModBlocks.blockFishTrap)) {
-                if (checkBlock != ModBlocks.blockFishTrap) {
```

Comment:
@Glease 
The broken logic stems from the fact that waterCount and trapCount aren't discretely compared and are pre-aggregated by incrementing water AND trap count when a trap block is present. This leads to the case where:
- 0 water blocks
- 2 trap blocks
would be internally tracked as:
- 2 water blocks
- 2 trap blocks

This is in addition to the string.contains("water") check which allows unintended blocks to count as water blocks and hurts performance by having to do a string.contains() six times per trap per 20 ticks.

Reply:
I see what you mean. Sorry for overlooking this chunk of the diff. However, I'm afriad this is intentional. I have sent a DM to alk (original author) and she confirms my theory. Fish catcher are also filled with water, so it counts as water.

Reply:
Thank you for clarifying the original intent! I will change the scope of this PR to only remove the string.contains and preserve the intended logic. Do you think it's worth updating the tool-tip to more accurately reflect the intended behavior? Proposing something like the following:

Current:
  list.add("This trap catches fish faster if surrounded by more water blocks");
  list.add("Can also be placed beside upto 4 other fish traps");
  list.add("Requires at least two faces touching water");
  list.add("1/1000 chance to produce triple loot.");
 
Proposed
  list.add("Requires at least two faces touching water or other fish catchers");
  list.add("Rates improve when surrounded by water");
  list.add("Rates also improve when surrounded by other fish catchers (max of four)");
  list.add("1/1000 chance per catch to produce triple loot.");